napari/utils/colormaps/colormap.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+from scipy import interpolate
+from enum import auto
+
+from .colorbars import make_colorbar
+from .standardize_color import transform_color
+from ...utils.dataclass import dataclass, Property
+from ...utils.misc import StringEnum
+
+
+class ColormapInterpolationMode(StringEnum):
+    """INTERPOLATION: Interpolation mode for colormaps."""
+
+    LINEAR = auto()
+    ZERO = auto()
+
+
+@dataclass(events=True, properties=True)
+class Colormap:
+    """Colormap that relates intensity values to colors.
+
+    Parameters
+    ----------
+    name : str
+        Name of the colormap.
+    data : array, shape (N, 4)
+        Data used in the colormap.
+    controls : array, shape (N,) or (N+1,)
+        Control points of the colormap.
+    interpolation : str
+        Colormap interpolation mode, either 'linear' or
+        'zero'. If 'linear', ncontrols = ncolors (one
+        color per control point). If 'zero', ncontrols
+        = ncolors+1 (one color per bin).
+    """
+
+    name: str = 'undefined'
+    colors: Property[np.ndarray, None, transform_color] = np.zeros((2, 4))
```

Comment:
suggest either removing colors and controls from the repr with [`field(repr=False)`](https://docs.python.org/3/library/dataclasses.html#dataclasses.field), or declare your own `__repr__`.